dm/dm.go

```diff
@@ -71,13 +72,15 @@ var usage = func() {
 	fmt.Fprintln(os.Stderr, "Flags:")
 	flag.PrintDefaults()
 	fmt.Fprintln(os.Stderr)
-	os.Exit(1)
+	fmt.Fprintln(os.Stderr, "--stdin requires a tar archive of input files and a top level file name.")
```

Comment:
The challenge is getting multiple files into the container with their
names, since the import mechanism requires the file names. If we don't use
tar, the user can only pass a single logical file on stdin. Yes, they can
catenary multiple files, but there's no way to separate them or name them
in the catenated stream. On the other hand, tar is a standard pattern for
working with containers, used by both Docker and Kubernetes. Users will
therefore find it natural. It's also quite easy to tar up a folder. tar cf
- myfolder | dm ...

On Friday, November 27, 2015, vaikas-google <notifications@github.com>
wrote:

> In dm/dm.go
> <https://github.com/kubernetes/deployment-manager/pull/131#discussion_r46074892>
> :
>
> > @@ -71,13 +72,15 @@ var usage = func() {
> >  	fmt.Fprintln(os.Stderr, "Flags:")
> >  	flag.PrintDefaults()
> >  	fmt.Fprintln(os.Stderr)
> > -	os.Exit(1)
> > +	fmt.Fprintln(os.Stderr, "--stdin requires a tar archive of input files and a top level file name.")
>
> This seems odd to require a tar as an input format instead of straight
> text. This means that to run say the guestbook example for a container
> (which I think is the driving use case) you'd need to tar the file up and
> then pass it in? Or did I misunderstand?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/deployment-manager/pull/131/files#r46074892>
> .
>


-- 
Jack Greenfield | Google